### PR TITLE
DEPR: enforce deprecations in core.internals

### PR DESCRIPTION
**Pull request by jbrockmendel**
Not sure if the whatsnew is needed for this; IIRC it was mainly pyarrow/fastparquet that were using these keywords

### CODE REVIEW

File: pandas/tests/internals/test_internals.py [hunk 1]
```diff
@@ -311,7 +311,8 @@ def test_delete(self):
     def test_make_block_same_class(self):
         # issue 19431
         block = create_block("M8[ns, US/Eastern]", [3])
-        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+        with pytest.raises(TypeError, match="unexpected keyword"):
```
Comment (WillAyd) [inline pandas/tests/internals/test_internals.py]: Should we just remove the dtype argument from this test instead of catching this?

Comment (jbrockmendel) [reply pandas/tests/internals/test_internals.py]: without the dtpye kwarg its just a smoke test, at which point id rather just remove this test (which id be completely fine with)

Comment (WillAyd) [reply pandas/tests/internals/test_internals.py]: I think OK then. As is this isn't testing anything pandas related just the stdlib

Comment (WillAyd) [reply pandas/tests/internals/test_internals.py]: *OK to remove

Comment (jbrockmendel) [reply pandas/tests/internals/test_internals.py]: removed

**Comment by jreback** [comment]
lgtm. rebase and merge.

**Comment by jbrockmendel** [comment]
the travis-36-locale build uses fastparquet=0.2.1 which apparently still passes dtype to make_block_same_class.  any objection to bumping the required version?

**Comment by jbrockmendel** [comment]
Looks like we last bumped the fastparquet version in #24590

**Comment by jorisvandenbossche** [comment]
Yes, I think bumping the minimum fastparquet version is fine. 
cc @TomAugspurger

**Comment by jreback** [comment]
> Yes, I think bumping the minimum fastparquet version is fine.
> cc @TomAugspurger

agreed here as well.

**Comment by TomAugspurger** [comment]
Happy to bump.

**Comment by jbrockmendel** [comment]
bumped fastparquet to 0.3.0.  Wasnt sure what to put in the whatsnew, so I copied the whole minimum-version section from 0.25.0, updated the fastparquet entry, and cut the sentence about py27 having just been dropped.

**Comment by jbrockmendel** [comment]
Looks like 0.3.2 is the version when fastparquet stopped using the deprecated keyword, but that is also the most recent version.  are we OK with bumping that far?

**Comment by jreback** [comment]
looks good, IIRC you have a nother PR where you add the Change column, can update then.

**Pull request merged**